distributed/bokeh/worker.py

```diff
@@ -422,6 +424,123 @@ def update(self):
+
+        if len(figures) <= 5:
+            self.root = column(figures, sizing_mode=sizing_mode)
+        else:
+            self.root = column(*[row(*pair, sizing_mode=sizing_mode)
+                                 for pair in partition_all(2, figures)],
+                               sizing_mode=sizing_mode)
+
+    def add_digest_figure(self, name):
+        with log_errors():
+            n = len(self.server.digests[name].intervals)
+            sources = {i: ColumnDataSource({'x': [], 'y': []})
+                        for i in range(n)}
+
+            kwargs = {}
+            if name.endswith('duration'):
+                kwargs['x_axis_type'] = 'datetime'
+
+            fig = figure(title=name, tools='', height=150,
+                    sizing_mode=self.sizing_mode, **kwargs)
+            fig.yaxis.visible = False
+            fig.ygrid.visible = False
+            if name.endswith('bandwidth') or name.endswith('bytes'):
+                fig.xaxis[0].formatter = NumeralTickFormatter(format='0.0b')
+
+            for i in range(n):
+                alpha = 0.3 + 0.3 * (n - i) / n
+                fig.line(source=sources[i], x='x', y='y',
+                         alpha=alpha, color=RdBu[max(n, 3)][-i])
+
+            fig.xaxis.major_label_orientation = math.pi / 12
+            fig.toolbar.logo = None
+            self.digest_sources[name] = sources
+            self.digest_figures[name] = fig
+            return fig
+
+    def add_counter_figure(self, name):
+        with log_errors():
+            n = len(self.server.counters[name].intervals)
+            sources = {i: ColumnDataSource({'x': [], 'y': [], 'y-center': []})
+                        for i in range(n)}
+
+            fig = figure(title=name, tools='', height=150,
+                    sizing_mode=self.sizing_mode,
+                    x_range=sorted(map(str, self.server.counters[name].components[0])))
+            fig.ygrid.visible = False
+
+            for i in range(n):
+                width = 0.5 + 0.4 * i / n
+                fig.rect(source=sources[i], x='x', y='y-center', width=width,
+                        height='y', alpha=0.3, color=RdBu[max(n, 3)][-i])
+                hover = HoverTool(
+                    point_policy="follow_mouse",
+                    tooltips="""@x : @counts"""
+                )
+                fig.add_tools(hover)
+                fig.xaxis.major_label_orientation = math.pi / 12
+
+            fig.toolbar.logo = None
+
+            self.counter_sources[name] = sources
+            self.counter_figures[name] = fig
+            return fig
+
+    def update(self):
+        with log_errors():
+            for name, figure in self.digest_figures.items():
+                digest = self.server.digests[name]
+                d = {}
+                for i, d in enumerate(digest.components):
+                    if d.size():
+                        try:
+                            xs = [d.quantile(i / 100) for i in range(1, 99)]
```

Comment:
Here's one for a normal distribution:

![hist2](https://cloud.githubusercontent.com/assets/2783717/21193105/ae6af2ae-c1f0-11e6-85a6-084227c87d94.png)

It would be simple enough to create a `hist` method on `TDigest` that takes in a number of bins, or the bin locations, and produces the `y`s for the centers. I may add this anyway.